src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -163,35 +158,31 @@ public User addUser(PrimaryKey id) {
 	}
 	
 	public ResponseEntity<?> getTopUsers(String selected, int page, int nums) {
-		List<User> users = new ArrayList<>();
-		if(selected.equals("checkout")) {
-			users = userRepository.findUserCheckouts(true);
+		List<TopUser> users = new ArrayList<>();
+		if(selected.equals("checkouts")) {
+			users = userRepository.getTopUserCheckouts(PageRequest.of(page, nums));
 		}
-		else if(selected.equals("faves")) {
-			users = userRepository.findUserFavourites(true);
+		else if(selected.equals("favs")) {
+			users = userRepository.getTopUserFavourites(PageRequest.of(page, nums));
 		}
 		else if(selected.equals("ratings")){
```

Comment:
What if we made these strings "checkouts", "favs", "ratings" into constants?

Reply:
At this point we do not have a global constants repository. Also, the strings will likely not change frequently nor it will be used elsewhere. So, I think it is safe to keep it as it is for now.